### PR TITLE
🐛 Move click listener back to the bookend

### PR DESCRIPTION
**Pull request by Enriqe**
When we moved the bookend to the drawer (#23761) we also moved the click listener to be outside of the bookend element shadow DOM. 

This prevents the clicked target from being forwarded correctly to the linker, since it would only see the shadow root.

This PR moves the click listener to be back inside the bookend shadow DOM.

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story-draggable-drawer.css [hunk 1]
```diff
@@ -108,11 +108,6 @@
   transition: none !important;
 }
 
-.i-amphtml-story-draggable-drawer-bookend .i-amphtml-story-draggable-drawer-content {
-  margin-top: auto !important;
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-draggable-drawer.css]: Is a bookend with no configuration still displayed at the bottom of the screen?
I can help you test if you upload a demo :))

**Comment by gmajoulet** [comment]
Could you please add a unit test too? Sorry I broke this.

**Comment by Enriqe** [comment]
Thanks, I was able to find a couple of bugs while doing the demo. 

I tried them now and everything seems to be working now, you can take a look here:

(page attachments 2nd page)
https://stamp-animation.firebaseapp.com/examples/amp-story/body-painting/index.html

(desktop 3 panels)
https://stamp-animation.firebaseapp.com/examples/amp-story/bookend.html

(desktop full bleed)
https://stamp-animation.firebaseapp.com/examples/amp-story/helloworld-bookend.html

(no bookend configured, only sharing options on mobile)
https://stamp-animation.firebaseapp.com

**Comment by gmajoulet** [comment]
I think we still have a couple issues with this implementation:
- You need to click on the 88px above the bookend to close it (if you click 89px higher it won't close)
- I also broke the link to re-prompt `amp-consent`, we should fix this with this PR too as they have the same root cause

Maybe instead of tweaking the layout (hard to test), we could tweak our JavaScript a bit.

I've been trying this instead, that adds another click listener on the shadow tree, it'd need more testing but maybe it can work?

```
    this.element.addEventListener('click', event =>
      this.onOuterShadowClick_(event)
    );

    this.getShadowRoot().addEventListener('click', event => {
      this.onInnerShadowClick_(event);
    });

  /**
   * @param {!Event} event
   * @private
   */
  onOuterShadowClick_(event) {
    const target = dev().assertElement(event.target);
    if (this.elementOutsideUsableArea_(target)) {
      event.stopPropagation();
      this.storeService_.dispatch(Action.TOGGLE_BOOKEND, false);
      return;
    }
  }

  /**
   * @param {!Event} event
   * @private
   */
  onInnerShadowClick_(event) {
    const target = dev().assertElement(event.target);

    event[AMP_CUSTOM_LINKER_TARGET] = target;

    if (target.hasAttribute('on')) {
      const actionService = Services.actionServiceForDoc(this.element);
      actionService.trigger(target, 'tap', event, ActionTrust.HIGH);
    }
  }
```

Thanks for looking into this!

**Comment by Enriqe** [comment]
Friendly ping

**Comment by Enriqe** [comment]
Np, thanks!

**Pull request merged**